### DISCUSSION TITLE
**FEATURE** Custom theming of RevoGrid

### DISCUSSION THREAD

**Discussion by Thoemmy** [revolist/revogrid#540] [General]
Story
As a RevoGrid Developer i want to create my custom theme for the revogrid table so that i can adjust the table to fit my UI styling.

Acceptance Criteria
- Table should be fully custom themeable
- Should be easy to config -e.g. through css tokens

This feature request is just a suggestion.

**Comment by revolist** [comment]
Hi @Thoemmy 
Do you want to be able to have a theme functionality where you would be able to pass css token colors and grid would be adopted accordingly?

**Reply by Thoemmy** [reply]
Hi @revolist,
yes exactly, but im not sure how this will impact your current logic with the themeManager.
But i think for future this will make less effort to provide finished themes for the useres cause they have the opportunity to create their own.

My Suggestion:
It could be provided as a themeManager plugin or rebuild scss structure and implementation and provide single theme scss files which also can be used to configure own styling.

Tokens could be like that:

```scss
/* custom-theme.scss */
$header-height: 50px;
$default-rows-size: 42px;
$header-size: 30px;
$header-bg: #f8f9fa;
$header-color: #000;
$header-border-color: #cecece;
$default-rows-size: 27px;
$cell-border-color: #e2e3e3;
$focused-header: rgba(#e9eaed, 0.5);

/* Not all tokens listed cause due too complexity of current implementation, 
example $cell-shadow-right is used for different stylings multiple times 
naming changes are required? */
```

**Reply by m2a2x** [reply]
Hi @Thoemmy 
I've been thinking about same improvement. 
We could do custom class which would be created per each theme
